frontend/apps/reader/modules/readergesture.lua

```diff
@@ -697,6 +701,20 @@ function ReaderGesture:gestureAction(action, ges)
         else
             self.ui:handleEvent(Event:new("SetScreenMode", "portrait"))
         end
+    elseif action == "toggle_wifi" then
+        local NetworkMgr = require("ui/network/manager")
+
+        if not NetworkMgr:isOnline() then
+            -- NB Normal widgets should use NetworkMgr:promptWifiOn()
+            -- This is specifically the toggle wifi action, so consent is implied.
+            NetworkMgr:turnOnWifi()
+        else
+            NetworkMgr:turnOffWifi()
+
+            UIManager:show(InfoMessage:new{
+                text = _("Wifi disabled."),
```

Comment:
No timeout to this InfoMessage? If we use gestures, it's that we want things fast and get done :) No need for this additional tap to dismiss the InfoMessage I think.
I would even use a Notification with timeout, for even less disruption.

Reply:
Right, good point.

I was testing this and had no idea if turning it off worked or not without going to the menu, which seemed to defeat the point.